tests/std/tests/P0674R1_make_shared_for_arrays/test.cpp

```diff
@@ -623,6 +623,67 @@ void test_allocate_shared_array_unknown_bounds() {
     }
 }
 
+// Test GH-1733 "<memory>: error C2694 when calling make_shared on class with throwing destructor"
+struct NontrivialThrowingDtor {
+    ~NontrivialThrowingDtor() noexcept(false) {}
+};
+static_assert(!is_nothrow_destructible_v<NontrivialThrowingDtor>);
+static_assert(!is_trivially_destructible_v<NontrivialThrowingDtor>);
+
+struct TrivialThrowingDtor {
+    ~TrivialThrowingDtor() noexcept(false) = default;
+};
+
+#ifndef __EDG__ // TRANSITION, VSO-1292292
+static_assert(!is_nothrow_destructible_v<TrivialThrowingDtor>);

```

Comment:
If the bug is that EDG thinks this type is nothrow destructible, I suggest adding an `#else` branch asserting the bug (with an appropriate comment) so we can't fail to notice when the bug is fixed.

Reply:
I can do that - however, it will `static_assert` when EDG is updated internally, in advance of that shipping in a public Preview. Not only will a non-STL dev encounter the `static_assert`, but we wouldn't be able to remove this workaround atomically. Therefore, I am inclined to leave this as-is, but I can make the change if you really want it.

Reply:
> I can do that - however, it will `static_assert` when EDG is updated internally, in advance of that shipping in a public Preview. Not only will a non-STL dev encounter the `static_assert`, but we wouldn't be able to remove this workaround atomically. Therefore, I am inclined to leave this as-is, but I can make the change if you really want it.

You've talked me out of it.